### PR TITLE
clean up public API doc

### PR DESCRIPTION
**Pull request by elharo**
@suztomo Trying to focus this on behavior rather than implementation in case we change the implementation for #1405

### CODE REVIEW

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java [hunk 1]
```diff
@@ -154,7 +155,7 @@ private DependencyNode resolveCompileTimeDependencies(
 
   /**
    * Finds the full compile time, transitive dependency graph including duplicates, conflicting
-   * versions, and dependencies with 'provided' scope.
+   * versions, and dependencies with provided and optional scope.
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java:158]: optional is not a scope. I call "optional flag".

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import static com.google.cloud.tools.opensource.dependencies.RepositoryUtility.CENTRAL;
 import static com.google.cloud.tools.opensource.dependencies.RepositoryUtility.mavenRepositoryFromUrl;
 import static com.google.common.collect.ImmutableList.toImmutableList;
-
+import com.google.common.annotations.VisibleForTesting;
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java:22]: Add one empty line between static import and non-static import.

Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java]: done

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java (hunk 1 above)
Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java]: true, rewrote

**Pull request merged**